### PR TITLE
Type hints: `utils` module

### PR DESCRIPTION
**Pull request by leandrodesouzadev**
typing: Adds type hints to the `utils` module. A `stubs` file was added with type definitions.

Partial implementation of #1705.

### CODE REVIEW

File: debug_toolbar/utils.py [hunk 1]
```diff
@@ -174,7 +177,7 @@ def get_name_from_obj(obj):
     return name
 
 
-def getframeinfo(frame, context=1):
+def getframeinfo(frame: Any, context=1) -> inspect.Traceback:
```
Comment (tim-schilling) [inline debug_toolbar/utils.py:180]: `context` was missed here.

File: debug_toolbar/utils.py [hunk 2]
```diff
@@ -131,7 +132,7 @@ def get_template_info():
     return template_info
 
 
-def get_template_context(node, context, context_lines=3):
+def get_template_context(node, context, context_lines=3) -> Dict[str, Any]:
```
Comment (tim-schilling) [inline debug_toolbar/utils.py:135]: The parameters here were missed.

**Review by tim-schilling — COMMENTED** [review]
It looks like `get_sorted_request_variable` needs a type hint as well.

File: debug_toolbar/utils.py (hunk 1 above)
Comment (leandrodesouzadev) [reply debug_toolbar/utils.py]: Just pushed a new commit.

File: debug_toolbar/utils.py (hunk 2 above)
Comment (leandrodesouzadev) [reply debug_toolbar/utils.py]: Done.

File: debug_toolbar/stubs.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from typing import Any, List, NamedTuple, Optional, Tuple
+
+from django import template as dj_template
```
Comment (tim-schilling) [inline debug_toolbar/stubs.py:3]: @matthiask Admittedly I haven't done typing in a project of my own that requires a stubs file. I'm unsure how I feel about a single stub file without docs or some way to determine if they're used someplace. However, I think I'm okay with us using this until we find it doesn't work.

Thoughts?

**Review by tim-schilling — APPROVED** [review]
I'm good with these changes, but want to see if we need a discussion regarding stubs.py before merging.

**Comment by matthiask** [comment]
I'm only wondering if the module should be renamed to `_stubs` to make it clear that it's not supposed to be used from outside django-debug-toolbar (not even by third party panels). At the very least I'd expect a comment stating that these stubs are purely internal.

Apart from that I have to admit that I haven't used typing on real world projects yet. I have played around with mypy a bit but that's all, so I can't really say if these changes make sense or not. FWIW I approve this pull request as well :)

**Comment by tim-schilling** [comment]
@leandrodesouzadev could you make the changes Matthias outlined? I think they're worthwhile.

**Comment by leandrodesouzadev** [comment]
@tim-schilling I just pushed a new commit with the changes outlined by @matthiask.

**Comment by matthiask** [comment]
Let's get starting.

**Comment by matthiask** [comment]
Thanks!

**Comment by sshishov** [comment]
You can just check guys how the types are added into other libraries

**Pull request merged**